### PR TITLE
feat(expect): update toBeInstanceOf

### PR DESCRIPTION
**Pull request by zhiyuang**
this is a follow up of previous pr: https://github.com/oven-sh/bun/pull/2389

1. use `isConstrutor()` replacing `jsType().isFunction()`
2. update `toBeInstanceOf` formatting message
3. update `isInstanceOf` binding ( C API -> C++ Binding), as for C API is not recommended anymore 🤔

### CODE REVIEW

File: src/bun.js/bindings/bindings.cpp [hunk 1]
```diff
@@ -3795,6 +3795,29 @@ bool JSC__JSValue__isConstructor(JSC__JSValue JSValue0)
...
+    JSValue jsValue = JSValue::decode(JSValue0);
+    JSValue jsValue1 = JSValue::decode(JSValue1);
+    if (UNLIKELY(!jsValue1.isObject())) {
+        return false;
+    }
+    JSObject* jsConstructor = JSC::asObject(jsValue1);
+    if (!jsConstructor->structure()->typeInfo().implementsHasInstance())
+        return false;
+    bool result = jsConstructor->hasInstance(globalObject, jsValue);
+    if (scope.exception()) {
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp]:
```suggestion
    RETURN_IF_EXCEPTION(scope, false);
```

File: src/bun.js/bindings/bindings.cpp [hunk 2]
```diff
@@ -3795,6 +3795,29 @@ bool JSC__JSValue__isConstructor(JSC__JSValue JSValue0)
...
+
+    auto scope = DECLARE_CATCH_SCOPE(vm);
+
+    JSValue jsValue = JSValue::decode(JSValue0);
+    JSValue jsValue1 = JSValue::decode(JSValue1);
+    if (UNLIKELY(!jsValue1.isObject())) {
+        return false;
+    }
+    JSObject* jsConstructor = JSC::asObject(jsValue1);
+    if (!jsConstructor->structure()->typeInfo().implementsHasInstance())
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp]:
```suggestion
    if (UNLIKELY(!jsConstructor->structure()->typeInfo().implementsHasInstance()))
```

**Comment by Jarred-Sumner** [comment]
Thank you for this

A couple nitpicky comments

**Review by Jarred-Sumner — APPROVED** [review]

**Comment by zhiyuang** [comment]
updated

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**